modules/node-shared/src/main/scala/org/tessellation/node/shared/infrastructure/consensus/message.scala

```diff
@@ -1,36 +1,44 @@
 package org.tessellation.node.shared.infrastructure.consensus
 
+import scala.annotation.nowarn
+
 import org.tessellation.node.shared.infrastructure.consensus.declaration.PeerDeclaration
 import org.tessellation.node.shared.infrastructure.consensus.declaration.kind.PeerDeclarationKind
 import org.tessellation.schema.peer.PeerId
 
-import derevo.cats.{eqv, show}
 import derevo.circe.magnolia.{decoder, encoder}
 import derevo.derive
 
 object message {
 
-  @derive(eqv, show, encoder, decoder)
+  @nowarn
```

Comment:
This is one of the places in the code (among others) where we get a false positive regarding the `encoder` and `decoder` being not used. We could alleviate it with `@nowarn` flag but it will at the same time suppress any future lawful warnings. I think I will resign from using the `@nowarn` for the mentioned reason in the end. If you have some idea on how we could alleviate it with no drawbacks, let me know.